File: part1-exercises-for-rust/hardways/leetcode-valid-parentheses/solution.c
bool isValid(char * s1){
        char s[100000];
        int i=0,j=0;
        for(j=0;s1[j]!=0;j++)
        {   
            char n = s1[j];
            if (n == '(')                s[i++] = ')';
            else if (n == '{')           s[i++] = '}';
            else if (n == '[')           s[i++] = ']';
            else 
            {
                if( i==0 || s[i-1] != n)      return false; 
                else        i--; 
            }        
        }
        return i==0;
}

File: part1-exercises-for-rust/hardways/leetcode-valid-parentheses/solution.rs
impl Solution {
    pub fn is_valid(s :String) -> bool {
        let mut stack = Vec::new();
        let mut flag = true;
        for i in s.chars() {
            if i == ')' {
                match stack.pop() {
                    None => {
                        flag = false;
                    },
                    Some(i) => {
                        if i != 0 {
                            flag = false;
                        }
                    }
                }
            }else if i == '('{
                stack.push(0);
            }else if i=='}'{
                 match stack.pop() {
                    None => {
                        flag = false;
                    },
                    Some(i) => {
                        if i != 1 {
                            flag = false;
                        }
                    }
                }
            }else if i == '{'{
                stack.push(1);
            }else if i==']'{
                 match stack.pop() {
                    None => {
                        flag = false;
                    },
                    Some(i) => {
                        if i != 2 {
                            flag = false;
                        }
                    }
                }
            }else if i=='['{
                stack.push(2);
            }
        }
        stack.len() == 0 && flag
    }
}